gateway/ln-gateway/src/lib.rs

```diff
@@ -1017,6 +1017,18 @@ impl Gateway {
             federation_info
         };
 
+        let balance = self
+            .handle_balance_msg(BalancePayload {
```

Comment:
Instead of reconstructing `BalancePayload`, you can just call `get_balance` on the client at line 1003